src/java.base/windows/native/libnio/ch/Net.c

```diff
@@ -727,7 +729,7 @@ Java_sun_nio_ch_Net_pollConnect(JNIEnv* env, jclass this, jobject fdo, jlong tim
                 NET_ThrowNew(env, lastError, "getsockopt");
             }
         } else if (optError != NO_ERROR) {
-            handleSocketError(env, optError);
+            NET_ThrowNew(env, optError, "getsockopt");
```

Comment:
This is not a `getsockopt` error, and using `"getsockopt"` here will make errors thrown here indistinguishable from errors thrown a few lines above. Can we use a different text?
The errors here will usually be the result of async connect, but I'm not sure if that's guaranteed; `SO_ERROR` would probably be the safest choice, but it doesn't look pretty. I guess that's one of the places where "no further information" actually makes sense.
Other than this, LGTM.